cpu/fe310/cpu.c

```diff
@@ -45,37 +51,59 @@ void cpu_init(void)
  *  cmd_code = 3
  *  all other fields = 0
  */
-
-__attribute__ ((section (".ramfunc")))
-void init_flash(void)
+void flash_init(void)
 {
-    /* Update the QSPI interface to adjust to the CPU speed
-     * This function needs to execute from the RAM
-     * when the QSPI interface is being reconfigured because the flash
-     * can't be accessed during this time
+    /* In case we are executing from QSPI, (which is quite likely) we need to
+     * set the QSPI clock divider appropriately before boosting the clock
+     * frequency.
      */
-
-    /* Disable SPI flash mode */
-    SPI0_REG(SPI_REG_FCTRL) &= ~SPI_FCTRL_EN;
-
-    /* Enable QPI mode by sending command to flash */
-    SPI0_REG(SPI_REG_TXFIFO) = 0x35;
+    SPI0_REG(SPI_REG_SCKDIV) = SCKDIV_SAFE;
 
     /* begin{code-style-ignore} */
-    SPI0_REG(SPI_REG_FFMT) =               /* setup "Fast Read Quad I/O (QPI mode)"         */
+    SPI0_REG(SPI_REG_FFMT) =               /* setup "Fast Read Dual I/O"                             */
         SPI_INSN_CMD_EN         |          /* Enable memory-mapped flash                    */
         SPI_INSN_ADDR_LEN(3)    |          /* 25LP03D read commands have 3 address bytes    */
-        SPI_INSN_PAD_CNT(6)     |          /* 25LP03D Table 6.11 Read Dummy Cycles = 6      */
-        SPI_INSN_CMD_PROTO(SPI_PROTO_Q) |  /* 25LP03D Table 8.1 "Instruction                */
-        SPI_INSN_ADDR_PROTO(SPI_PROTO_Q) | /*  Set" shows mode for cmd, addr, and           */
-        SPI_INSN_DATA_PROTO(SPI_PROTO_Q) | /*  data protocol for given instruction          */
-        SPI_INSN_CMD_CODE(0xEB) |          /* Set the instruction to "Fast Read Quad I/O"   */
+        SPI_INSN_PAD_CNT(4)     |          /* 25LP03D Table 6.11 Read Dummy Cycles = 4      */
```

Comment:
In IS25LP128 this is table 6.10

Reply:
This is only upto 104Mhz, If I understand the table correctly. Otherwise it would be 8.

Reply:
Not resolved but posponed